analyzers/tests/SonarAnalyzer.UnitTest/CFG/Roslyn/BasicBlockTest.cs

```diff
@@ -118,6 +124,88 @@ public void Method(bool condition)
+        public void ValidateOperations()
+        {
+            const string code = @"
+public class Sample
+{
+    int Pow(int num, int exponent)
+    {
+        num = num * Pow(num, exponent - 1);
+        return 42;
+    }
+}";
+            var cfg = TestHelper.CompileCfg(code);
+            var entry = cfg.EntryBlock;
+            var exit = cfg.ExitBlock;
+            var body = cfg.Blocks[1];
+            entry.Kind.Should().Be(BasicBlockKind.Entry);
+            entry.Operations.Should().BeEmpty();
+            exit.Kind.Should().Be(BasicBlockKind.Exit);
+            exit.Operations.Should().BeEmpty();
+            body.Kind.Should().Be(BasicBlockKind.Block);
+            body.Operations.Should().HaveCount(1).And.Subject.Single().Should().BeAssignableTo<IExpressionStatementOperation>();
+            body.BranchValue.Should().BeAssignableTo<ILiteralOperation>();
+            body.OperationsAndBranchValue.Should().OnlyContainInOrder(body.Operations[0], body.BranchValue);
+        }
+
+        [TestMethod]
+        public void ValidateSwitchExpressionCase()
+        {
+            const string code = @"
+public class Sample
+{
+    public int Method(bool condition) =>
+        condition switch
+        {
+            true => 42,
+            _ => 43
+        };
+}";
+            var cfg = TestHelper.CompileCfg(code);
+            /*
+             *
+             *         Block 1
+             *        true => 42
+             * Else  /          \    WhenFalse
+             *      /            \
+             *     /              \
+             *  Block 2          Block 3
+             *   => 42           _ => 43
+             *    |                |            WhenFalse from Block 3 - Block 5 should not be reachable
+             *    |                |
+             *    |              Block 4        Block 5
+             *    |               => 43         no match => throw exception
+             *    \                /
+             *     \              /
+             *      \            /  Else
+             *       \          /
+             *        \        /
+             *          Block 6
+             */
+            var block1 = cfg.Blocks[1];
+            var block2 = cfg.Blocks[2];
+            var block3 = cfg.Blocks[3];
+            var block4 = cfg.Blocks[4];
+            var block5 = cfg.Blocks[5];
+            var block6 = cfg.Blocks[6];
+            block1.FallThroughSuccessor.Destination.Should().Be(block2);
+            block1.ConditionalSuccessor.Destination.Should().Be(block3);
+            block1.SuccessorBlocks.Should().ContainInOrder(block2, block3);
+            block2.FallThroughSuccessor.Destination.Should().Be(block6);
+            block2.SuccessorBlocks.Single().Should().Be(block6);
+            block3.FallThroughSuccessor.Destination.Should().Be(block4);
+            block3.ConditionalSuccessor.Destination.Should().Be(block5);
+            block3.SuccessorBlocks.Single().Should().Be(block4);
```

Comment:
Let's add a comment for the irregularity
```suggestion
            block3.SuccessorBlocks.Single().Should().Be(block4);	// We don't add the unreachable ConditionalSuccessor in this case
```